### ISSUE TITLE
Carthage bootstrap step runs for all lanes

### ISSUE THREAD

**Issue by eneko** [fastlane/fastlane#10887]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I have a repo with Carthage. It is nice to see Fastlane running `carthage bootstrap` when building the app.

However, the Carthage step is run every time, for every lane, even if the lane has nothing to do with the project or Xcode.

For instance, consider this lane:
```
desc "Display number of commits"
lane :commit_number do
  puts number_of_commits
end
```

It will still run the `carthage bootstrap` step.

I would like to know a way to **skip the Carthage step entirely** (if the repo is already bootstrapped, for instance). **A way to configure it would be great too**, for instance, specifying to always use the cache: `carthage bootstrap --cache-builds` to use cached builds when possible.

##### Complete output when running fastlane, including the stack trace and command used
```
$ fastlane commit_number
[08:37:53]: fastlane detected a Gemfile in the current directory
[08:37:53]: however it seems like you don't use `bundle exec`
[08:37:53]: to launch fastlane faster, please use
[08:37:53]: 
[08:37:53]: $ bundle exec fastlane commit_number
[08:37:53]: 
[08:37:53]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[08:37:56]: -------------------------------------------------
[08:37:56]: --- Step: Verifying required fastlane version ---
[08:37:56]: -------------------------------------------------
[08:37:56]: Your fastlane version 2.64.0 matches the minimum requirement of 2.64.0  ✅
[08:37:56]: ------------------------------
[08:37:56]: --- Step: default_platform ---
[08:37:56]: ------------------------------
[08:37:56]: Driving the lane 'ios commit_number' 🚀
[08:37:56]: ----------------------
[08:37:56]: --- Step: carthage ---
[08:37:56]: ----------------------
[08:37:56]: $ carthage bootstrap
...
[08:40:23]: -------------------------------
[08:40:23]: --- Step: number_of_commits ---
[08:40:23]: -------------------------------
[08:40:23]: $ git rev-parse HEAD
[08:40:23]: ▸ 102c00e8177d8e93f311a960f4a10f5bd38b22f8
[08:40:23]: $ git rev-list HEAD --count
[08:40:23]: ▸ 97
[08:40:24]: 97

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | carthage                            | 147         |
| 4    | number_of_commits                   | 0           |
+------+-------------------------------------+-------------+

[08:40:24]: fastlane.tools finished successfully 🎉
```

TLDR: Getting the number of commits took almost 2 minutes and 30 seconds.

**Comment by eneko** [comment]
Adding a manual Carthage step to the lane seems to run Carthage twice:
```
desc "Display number of commits"
lane :commit_number do
  carthage(cache_builds: true)
  puts number_of_commits
end
```

**Comment by eneko** [comment]
Nevermind, I must be blind. Seems like the autogenerated `Fastfile` included a `carthage` step in 'before_all':
```
before_all do
  # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
  # cocoapods
  carthage
end
```

Probably a bad idea to add that as a default during set up, since it might not be required for all lanes.
